lib/public/views/lhcPeriods/Overview/LhcPeriodsOverviewModel.js

```diff
@@ -55,38 +60,20 @@ export class LhcPeriodsOverviewModel extends Observable {
      * @returns {void}
      */
     async fetchLhcPeriods() {
-        const keepExisting = this._pagination.currentPage > 1 && this._pagination.isInfiniteScrollEnabled;
-
-        if (!keepExisting) {
-            this._lhcPeriods = RemoteData.loading();
-            this.notify();
-        }
-
         const queryParams = {
             'page[offset]': this._pagination.firstItemOffset,
             'page[limit]': this._pagination.itemsPerPage,
         };
-
         const { appliedOn: sortOn, appliedDirection: sortDirection } = this._overviewSortModel;
         if (sortOn && sortDirection) {
             queryParams[`sort[${sortOn}]`] = sortDirection;
         }
         const endpoint = buildUrl('/api/lhcPeriodsStatistics', { ...queryParams, ...this.getFilteringQueryParameters() });
 
-        try {
-            const { items, totalCount } = await getRemoteDataSlice(endpoint);
-            const concatenateWith = keepExisting ? this._lhcPeriods.payload || [] : [];
-            this._lhcPeriods = RemoteData.success([
-                ...concatenateWith,
-                ...items.map((lhcPeriodData) => this.flattenPeriodDataObject(lhcPeriodData)),
-            ]);
-
-            this._pagination.itemsCount = totalCount;
-        } catch (errors) {
-            this._lhcPeriods = RemoteData.failure(errors);
-        }
-
-        this.notify();
+        await this._remoteDataFetcher.fetch(endpoint, {
+            keepExisting: this._pagination.currentPage > 1 && this._pagination.isInfiniteScrollEnabled,
```

Comment:
For me, the choice of keeping existing or not should be where concatenation is done